client/components/number-field/number-input.js

```diff
@@ -1,44 +1,38 @@
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import FormTextInput from 'components/forms/form-text-input';
 
-export default React.createClass( {
-	displayName: 'NumberInput',
-
-	propTypes: {
+export default class NumberInput extends Component {
+	static propTypes = {
 		onChange: PropTypes.func,
-	},
+	};
 
-	getDefaultProps() {
-		return {
-			onChange: () => {},
-		};
-	},
+	static defaultProps = {
+		onChange: () => {},
+	};
 
-	getInitialState() {
-		return {
-			focused: false,
-			text: this.props.value,
-		};
-	},
+	state = {
+		focused: false,
+		text: this.props.value,
+	};
```

Comment:
I'm sure this is equivalent, but the [React docs](https://facebook.github.io/react/docs/state-and-lifecycle.html#adding-local-state-to-a-class) and components within Calypso set state in the constructor.